common/src/main/java/com/lightstep/tracer/shared/AbstractTracer.java

```diff
@@ -259,7 +259,8 @@ private void maybeStartReporting() {
         reportingThread.setDaemon(true);
         reportingThread.start();
 
-        if (!disableMetricsReporting && safeMetrics != null) {
+        if (!disableMetricsReporting && safeMetrics != null
```

Comment:
My impression is that the `disableMetricsReporting` attribute is the one that should be directly set from `Options`, so we should be able to override it there.

```java
  // Options
  private boolean disableMetricsReporting = getEnvMetricsEnabled();

  static boolean getEnvMetricsEnabled() {
    String envMetricsEnabled = System.getenv(...);
    if (envMetricsEnabled != null) {
      // Return true/false if parsed properly.
      // Else, log the invalid value, and...
    }
    // ... Return DEFAULT_DISABLE_METRICS; // false
  }
```

Reply:
updated